azDevOps/azure/templates/jobs/deploy-helm.yml

```diff
@@ -0,0 +1,81 @@
+parameters:
+  kubernetes_service_connection: ''
+  registry_name: ''
+  registry_login: ''
+  registry_pass: ''
+  chart_name: ''
+  namespace: ''
+  overrides: ''
+  helm_version: 3.0.2
+  acr_service_connection: ''
+  functional_test: false
+  functional_test_artefact_name: ''
+  functional_test_project_file: ''
+  functional_test_config_override: false
+  functional_test_config_overrides: ''
+
+jobs:
+  - job: Helm_Deployment
+    pool:
+      vmImage: 'ubuntu-18.04'
+    steps:
+      - task: HelmInstaller@0
+        inputs:
+          connectionType: 'Kubernetes Service Connection'
+          kubernetesServiceConnection: ${{parameters.kubernetes_service_connection}}
+          kubernetesCluster: ${{parameters.cluster_name}}
+          helmVersion: ${{parameters.helm_version}}
+          checkLatestHelmVersion: false
+        displayName: 'Helm: Install ${{parameters.helm_version}}'
+
+      - task: AzureCLI@2
+        inputs:
+          azureSubscription: ${{parameters.acr_service_connection}}
+          scriptType: 'bash'
+          scriptLocation: 'inlineScript'
+          inlineScript: |
+            az acr helm repo add --name ${{parameters.registry_name}}
+            helm repo update
+            helm repo list
+        displayName: 'Helm: Add Registry'
+
+      - task: HelmDeploy@0
+        inputs:
+          connectionType: 'Kubernetes Service Connection'
+          kubernetesServiceConnection: ${{parameters.kubernetes_service_connection}}
+          namespace: ${{parameters.namespace}}
+          command: upgrade
+          chartName: '${{parameters.registry_name}}/${{parameters.chart_name}}'
+          releaseName: ${{parameters.chart_name}}
+          overrideValues: $${{parameters.overrides}}
+          force: false
+          arguments: '--version $(Build.BuildNumber) --set image.repository=${{parameters.registry_name}}.azurecr.io/${{parameters.chart_name}} --set image.tag=$(Build.BuildNumber)'
+        displayName: 'Helm: Upgrade-Install'
+
+      # Functional Testing

```

Comment:
Would be better to split the tests I to separate template and reuse with deploy-dotnet